src/Chainweb/Pact/PactService.hs

```diff
@@ -866,21 +851,38 @@ execLookupPactTxs restorePoint txs
     | V.null txs = return mempty
     | otherwise = go
   where
-    getRestorePoint cp =
-        case restorePoint of
-            Nothing -> do
-                (fmap (\(h, a) -> T2 h a)) <$> liftIO (_cpGetLatestBlock cp)
-            x -> return x
+    getRestorePoint = case restorePoint of
+        Nothing -> fmap (\bh -> T2 (_blockHeight bh) (_blockHash bh))
+            <$> findLatestValidBlock
+        x -> return x
 
     go = do
         cp <- getCheckpointer
-        mrp <- getRestorePoint cp
+        mrp <- getRestorePoint
         case mrp of
             Nothing -> return mempty      -- can't look up anything at genesis
             Just (T2 lh lha) ->
-                withCheckpointerRewind (Just (lh + 1, lha)) "lookupPactTxs" $
-                \_pdbenv -> fmap Discard $ liftIO $
-                V.mapM (_cpLookupProcessedTx cp) txs
+                withCheckpointerRewind (Just (lh + 1, lha)) "lookupPactTxs" $ \_ ->
+                    liftIO $ Discard <$> V.mapM (_cpLookupProcessedTx cp) txs
+
+findLatestValidBlock :: PactServiceM cas (Maybe BlockHeader)
+findLatestValidBlock = getCheckpointer >>= liftIO . _cpGetLatestBlock >>= \case
+    Nothing -> return Nothing
+    Just (height, hash) -> go height hash
+  where
+    go height hash = do
+        bhdb <- view psBlockHeaderDb
+        liftIO (lookup bhdb hash) >>= \case
+            Nothing -> do
+                logInfo $ "Latest block isn't valid."
```

Comment:
If we expect this case to be common we should probably reduce priority from `Info`